tests/providers/slack/hooks/test_slack.py

```diff
@@ -19,82 +19,154 @@
 import unittest
 
 import mock
+from slack.errors import SlackApiError
 
 from airflow.exceptions import AirflowException
 from airflow.providers.slack.hooks.slack import SlackHook
 
 
 class TestSlackHook(unittest.TestCase):
-    def test_init_with_token_only(self):
+
+    def test___get_token_with_token_only(self):
+        """ tests `__get_token` method when only token is provided """
+        # Given
         test_token = 'test_token'
-        slack_hook = SlackHook(token=test_token, slack_conn_id=None)
+        test_conn_id = None
+
+        # Creating a dummy subclass is the easiest way to avoid running init
+        #  which is actually using the method we are testing
+        class DummySlackHook(SlackHook):
+            def __init__(self, *args, **kwargs):  # pylint: disable=W0231
+                pass
 
-        self.assertEqual(slack_hook.token, test_token)
+        # Run
+        hook = DummySlackHook()
+
+        # Assert
+        output = hook._SlackHook__get_token(test_token, test_conn_id)  # pylint: disable=E1101
+        expected = test_token
+        self.assertEqual(output, expected)
 
     @mock.patch('airflow.providers.slack.hooks.slack.SlackHook.get_connection')
-    def test_init_with_valid_slack_conn_id_only(self, get_connection_mock):
+    def test___get_token_with_valid_slack_conn_id_only(self, get_connection_mock):
+        """ tests `__get_token` method when only connection is provided """
+        # Given
+        test_token = None
+        test_conn_id = 'x'
         test_password = 'test_password'
+
+        # Mock
         get_connection_mock.return_value = mock.Mock(password=test_password)
 
-        test_slack_conn_id = 'test_slack_conn_id'
-        slack_hook = SlackHook(token=None, slack_conn_id=test_slack_conn_id)
+        # Creating a dummy subclass is the easiest way to avoid running init
+        #  which is actually using the method we are testing
+        class DummySlackHook(SlackHook):
+            def __init__(self, *args, **kwargs):  # pylint: disable=W0231
+                pass
```

Comment:
I mean creating SlackHook is fine isn't it ? As the WebClient does not call any external service by itself until `WebClient.call` is run

Reply:
I want to test the `get_token` which is called in the `__init__` so this is why I'm creating a dummy subclass with empty init. Is there a better way to do it?

Reply:
There is no need of dummy subclasses, you can initialize the Slack directly, it is not calling slack API in the __init__ is it?

Reply:
Pushed a commit to PR: https://github.com/apache/airflow/pull/5519/commits/d81d6e947ba364b0dd9e730507e51ff2a0d3d3e2